src/hooks/useRecentConnection.tsx

```diff
@@ -0,0 +1,41 @@
+import { atomWithStorage } from 'jotai/utils';
+import { atom, useAtom } from 'jotai';
+import { SignClientTypes } from '@walletconnect/types';
+
+interface MetadataWithDate extends SignClientTypes.Metadata {
+  date: Date;
+}
+
+type StoredValueType = {
+  [url: string]: MetadataWithDate
+}
```

Comment:
For the linter issue
```suggestion
type StoredValueType = Record<string, MetadataWithDate>;
```